worldedit-forge/src/main/java/com/sk89q/worldedit/forge/ForgeWorldEdit.java

```diff
@@ -152,56 +154,57 @@ public void onPlayerInteract(PlayerInteractEvent event) {
             return;
         }
 
-        if (!platform.isHookingEvents()) return; // We have to be told to catch these events
+        if (!platform.isHookingEvents())
+            return; // We have to be told to catch these events
 
-        if (event.useItem == Result.DENY || event.entity.worldObj.isRemote) return;
+        if ((event instanceof PlayerInteractEvent.LeftClickBlock
+                && ((PlayerInteractEvent.LeftClickBlock) event)
+                        .getUseItem() == Result.DENY)
+                || (event instanceof PlayerInteractEvent.RightClickBlock
+                        && ((PlayerInteractEvent.RightClickBlock) event)
+                                .getUseItem() == Result.DENY)
+                || event.getEntity().worldObj.isRemote)
+            return;
```

Comment:
Maybe you should break this out into several condition booleans
```java
boolean isLeftClickBlock = ...;
boolean isLeftClickDenied = isLeftClickBlock && ...;
boolean isRightClickBlock = ...;
...
```

That would make these if statements look much nicer.